### PR TITLE
Allow access to webhook server from shoot API servers

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
This extension now uses the simplified `NetworkPolicy` approach for allowing traffic to its webhook server from `kube-apiserver`s of shoot clusters.

**Special notes for your reviewer**:
Similar to https://github.com/gardener/gardener/pull/8076 for `provider-local`

**Release note**:

```feature developer
This extension now uses the simplified `NetworkPolicy` approach for allowing traffic to its webhook server from `kube-apiserver`s of shoot clusters.
```

### CODE REVIEW

File: charts/gardener-extension-networking-cilium/templates/service.yaml [hunk 1]
```diff
@@ -6,7 +6,8 @@ metadata:
   annotations:
     networking.resources.gardener.cloud/from-world-to-ports: '[{"protocol":"TCP","port":{{ .Values.webhookConfig.serverPort }}}]'
     networking.resources.gardener.cloud/from-all-seed-scrape-targets-allowed-ports: '[{"port":{{ .Values.metrics.port }},"protocol":"TCP"}]'
-    networking.resources.gardener.cloud/namespace-selectors: '[{"matchLabels":{"kubernetes.io/metadata.name":"garden"}}]'
+    networking.resources.gardener.cloud/from-all-webhook-targets-allowed-ports: '[{"protocol":"TCP","port":{{ .Values.webhookConfig.serverPort }}}]'
+    networking.resources.gardener.cloud/namespace-selectors: '[{"matchLabels":{"kubernetes.io/metadata.name":"garden"}},{"matchLabels":{"kubernetes.io/metadata.name":"shoot"}}]'
```
Comment (ScheererJ) [inline charts/gardener-extension-networking-cilium/templates/service.yaml:10]: Is there some magic involved here or should this rather look like the following?

```suggestion
    networking.resources.gardener.cloud/namespace-selectors: '[{"matchLabels":{"kubernetes.io/metadata.name":"garden"}},{"matchLabels":{"gardener.cloud/role:":"shoot"}}]'
```

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: charts/gardener-extension-networking-cilium/templates/service.yaml (hunk 1 above)
Comment (rfranzke) [reply charts/gardener-extension-networking-cilium/templates/service.yaml]: 😵 Definitely, nice catch

Comment (ScheererJ) [reply charts/gardener-extension-networking-cilium/templates/service.yaml]: I would even argue that https://github.com/gardener/gardener/pull/8076 is also incorrect. The documentation part is fine, but the service is not.

**Review by ScheererJ — APPROVED** [review]
Thanks a lot for the change.

/lgtm

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Pull request merged**